apprating/src/main/java/com/mindera/skeletoid/apprating/controller/AppRatingController.kt

```diff
@@ -0,0 +1,124 @@
+package com.mindera.skeletoid.apprating.controller
+
+import android.content.Context
+import com.mindera.skeletoid.apprating.callbacks.AppRatingDialogResponse
+import com.mindera.skeletoid.apprating.callbacks.AppRatingDialogResponseCallback
+import com.mindera.skeletoid.apprating.job.AppRatingJobInitializer
+import com.mindera.skeletoid.apprating.store.AppRatingStore
+import com.mindera.skeletoid.apprating.utils.DateUtils
+import java.util.Date
+
+/**
+ * Controller responsible for handling the stored data and the dialog itself
+ */
+class AppRatingController {
+
+    private var countsPerTimeInterval: Pair<Int, Long>? = null
+    private var promptTimeInterval: Long? = null
+
+    /**
+     * Sets up the conditions to prompt the dialog.
+     *
+     * @param countsPerTimeInterval Pair<Int, Long> values with the maximum number of times the dialog can be prompt per time range (in days)
+     * @param promptTimeInterval Time distance between prompts (in days)
+     */
+    fun setupConditions(countsPerTimeInterval: Pair<Int, Long>, promptTimeInterval: Long) {
+        this.countsPerTimeInterval = countsPerTimeInterval
+        this.promptTimeInterval = promptTimeInterval
+    }
+
+    /**
+     * Checks if the data in the store matches the conditions and updates the values in the store.
+     *
+     * @param context Context
+     * @param dialogResultCallback Callback to handle default dialog response
+     * @return true if the conditions were met, and the dialog can be prompted.
+     */
+    fun promptDialog(context: Context, dialogResultCallback: AppRatingDialogResponseCallback? = null): Boolean {
+        val store = AppRatingStore(context)
+        return shouldPromptDialog(store).also {
+            if (it) {
+                dialogResultCallback?.let {
+                    TODO("IMPLEMENT DEFAULT DIALOG")
+                }
+                store.promptedCount++
+                store.lastTimePrompted = DateUtils.formatDate(Date())
```

Comment:
Should we be careful with dates? Because of timezones and etc ?

Reply:
We are calculating the diff between day by using the `time` from a Date. This already takes in consideration the GMT time zone (from what it says in the docs)

Reply:
I might be wrong but I think that only the initial date (Jan. 1, 19XX) is in GMT which is used to calculate the milliseconds between there and now.

Reply:
Added time zone to the `parse` and `format` in the `DateUtils` class